modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -237,6 +238,16 @@ const Modal = ({
 			eventHandlers.push(selectEventHandler);
 		}
 
+		customEvents.forEach((customEvent) => {
+			if (customEvent.name && customEvent.onEvent) {
+				const eventHandler = Liferay.on(customEvent.name, (event) => {
+					customEvent.onEvent(event);
+				});
```

Comment:
I don't know off the top of my head if `Liferay.on` passes more than just `event` to its callbacks, but this is probably equivalent:

```
const eventHandler = Liferay.on(customEvent.name,  customEvent.onEvent);
```